File: SourceCode/PantryFrontend/pantry-app/src/lib/modules/goods/types/Good.ts
export type GoodDetails = {
	tags: string[];
	purchaseLocations: string[];
	ratings: number[];
	totalPurchaseNumber: number;
	daysUntilConsume: number | null;
};

export type Good = {
	id: string;
	name: string;
	description: string | null;
	amount: number;
	minimumAmount: number | null;
	storageLocation: string | null;
	ean: number | null;
	currentPrice: number | null;
	shoppinglistName: string | null;
	details: GoodDetails;
	unitOfMeasurement: UnitOfMeasurement;
	//priceHistories: any[];
};

export type GoodOverview = {
	id: string;
	name: string;
	description: string;
	tags: string[];
	rating: number | null;
};

export type Goods = {
	goods: Good[];
};

export type GoodCreateDto = {
	name: string;
};

export type GoodRatingCreateDto = {
	goodId: string;
	rating: number;
};

export interface GoodSuggestion {
	id: string;
	name: string;
	unitOfMeasurement: UnitOfMeasurement;
}

export enum UnitOfMeasurement {
	Milliliter = 0,
	Gram = 1,
	Piece = 2
}

export const UnitOfMeasurementDisplayName = {
	[UnitOfMeasurement.Milliliter]: 'Milliliter (ml)',
	[UnitOfMeasurement.Gram]: 'Gramm (g)',
	[UnitOfMeasurement.Piece]: 'Stück (Stk)'
};

export function getUnitOfMeasurementDisplayName(unitOfMeasurement: UnitOfMeasurement): string {
	return UnitOfMeasurementDisplayName[unitOfMeasurement];
}

File: SourceCode/PantryBackend/Pantry.Shared/Models/GoodModels/Good.cs
namespace Pantry.Shared.Models.GoodModels;
public class Good
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public double Amount { get; set; }
    public double? MinimumAmount { get; set; }
    public string? StorageLocation { get; set; }
    public long? EAN { get; set; }
    public double? CurrentPrice { get; set; }
    public string? ShoppinglistName { get; set; }
    public GoodDetails Details { get; set; } = new GoodDetails();
    public List<PriceHistory> PriceHistories { get; set; } = new();
}
